src/server/req_modify.c

```diff
@@ -862,6 +859,28 @@ req_modifyReservation(struct batch_request *preq)
 					req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
 					return;
 				}
+
+				break;
+			case RESV_ATR_duration:
+				if (presv->ri_alter_etime == 0) {
+					temp = strtol(psatl->al_value, &end, 10) + presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+					if (!is_standing || temp < next_occr_start) {
+						presv->ri_alter_flags |= RESV_START_TIME_MODIFIED;
+						presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+					}
+					else {
+						snprintf(log_buffer, sizeof(log_buffer), "%s", msg_stdg_resv_occr_conflict);
+						log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
+							preq->rq_ind.rq_modify.rq_objname, log_buffer);
+						req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
+						return;
+					}
+				}
+				if ((!is_standing || temp < next_occr_start) && presv->ri_alter_stime == 0)
+					presv->ri_alter_stime = presv->ri_wattr[RESV_ATR_start].at_val.at_long;			
```

Comment:
I'm not sure you can do this in the while loop. I think you need to do this check before the `if (send_to_scheduler)`. At that point, we know all the values that are going to change.

Reply:
Also, you shouldn't have to set ri_alter_stime or etime, it will be set in their respective ATTR cases.

Reply:
I am setting this for standing reservations, if a backup is not present the standing reservation sets the new time for next occurances

Reply:
I am doing this for standing reservations, in case alter_etime or alter_stime is not set the time for next occurrence also changes

Reply:
Also you are right, I am moving this block of code outside while loop